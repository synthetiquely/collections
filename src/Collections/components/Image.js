import React from 'react';
import glamorous from 'glamorous';

const View = glamorous.div(
  {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    ':hover': {
      boxShadow: '1px 1px 1px 1px #ccc',
    },
  },
  ({ orientation }) => {
    if (orientation === 'landscape') {
      return {
        gridColumn: 'span 2',
      };
    } else if (orientation === 'portrait') {
      return {
        gridRow: 'span 2',
      };
    }
    return {
      gridColumn: 'span 2',
      gridRow: 'span 2',
    };
  },
);

const Image = glamorous.img({
  width: '100%',
  height: '100%',
  objectFit: 'cover',
});

export default props => (
  <View orientation={props.orientation} onClick={props.onClick}>
    <Image src={props.src} alt={props.title} />
  </View>
);
