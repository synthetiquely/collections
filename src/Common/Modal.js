import React from 'react';
import glamorous from 'glamorous';

const Backdrop = glamorous.div({
  position: 'fixed',
  display: 'flex',
  width: '100%',
  height: '100%',
  top: '0px',
  left: '0px',
  zIndex: '1',
  backgroundColor: 'rgba(0,0,0,0.6)',
});

const Window = glamorous.div({
  width: '80vh',
  height: '80vh',
  zIndex: '2',
  padding: '30px',
  display: 'flex',
  margin: 'auto',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f6f5f3',
  '@media only screen and (max-width: 1024px) and (orientation: portrait)': {
    width: '80vw',
    height: '80vw',
  },
  '@media only screen and (max-width: 1024px) and (orientation: landscape)': {
    width: '80vh',
    height: '80vh',
  },
});

const CloseButton = glamorous.span({
  position: 'absolute',
  top: '-5px',
  right: '5px',
  color: '#aaa',
  fontSize: '28px',
  fontWeight: 'bold',
  cursor: 'pointer',
});

const Next = glamorous.div({
  zIndex: '3',
  position: 'absolute',
  top: '50%',
  right: '0',
  width: 'auto',
  marginTop: '-22px',
  padding: '10px',
  color: '#fff',
  fontWeight: 'bold',
  fontSize: '20px',
  transition: '0.6s ease',
  borderRadius: '0 3px 3px 0',
  cursor: 'pointer',
  ':hover,:active,:focus': {
    fontSize: '25px',
  },
  '@media only screen and (max-width: 500px) and (orientation: portrait)': {
    top: '90%',
    right: '50%',
    left: '50%',
    margin: '0',
    padding: '5px',
    transform: 'rotateZ(90deg)',
  },
});

const Prev = glamorous.div({
  zIndex: '3',
  position: 'absolute',
  top: '50%',
  width: 'auto',
  left: '0',
  marginTop: '-22px',
  padding: '10px',
  color: '#fff',
  fontWeight: 'bold',
  fontSize: '20px',
  transition: '0.6s ease',
  borderRadius: '3px 0 0 3px',
  cursor: 'pointer',
  ':hover,:active,:focus': {
    fontSize: '25px',
  },
  '@media only screen and (max-width: 500px) and (orientation: portrait)': {
    top: '0',
    left: '50%',
    margin: '0',
    padding: '5px',
    transform: 'rotateZ(90deg)',
  },
});

const Modal = (props) => {
  if (props.isOpen) {
    return (
      <React.Fragment>
        <Backdrop onClick={props.onClose}>
          <Window>
            <CloseButton onClick={props.onClose}>&times;</CloseButton>
            <Next onClick={props.onNext}>&#10095;</Next>
            <Prev onClick={props.onPrevious}>&#10094;</Prev>
            {props.children}
          </Window>
        </Backdrop>
      </React.Fragment>
    );
  }
  return null;
};

export default Modal;
