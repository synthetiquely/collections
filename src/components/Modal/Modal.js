import React from 'react';
import { DESTINATION_NEXT, DESTINATION_PREVIOUS } from '../../constants';
import Backdrop from '../styled/Backdrop';
import ModalWindow from '../styled/ModalWindow';
import ModalContent from '../styled/ModalContent';
import ArrowButton from '../styled/ArrowButton';
import CloseButton from '../styled/CloseButton';
import Icon from '../styled/Icon';

const Modal = props => (
  <Backdrop>
    <ModalWindow id="preview-modal">
      <CloseButton id="preview-close-button" onClick={props.onClose}>
        <Icon size="lg">&times;</Icon>
      </CloseButton>
      <ArrowButton
        id="preview-right-arrow"
        direction="right"
        onClick={() => props.onChangeSelected(DESTINATION_NEXT)}
      >
        <Icon rotatable size="md">
          &#10095;
        </Icon>
      </ArrowButton>
      <ArrowButton
        id="preview-left-arrow"
        direction="left"
        onClick={() => props.onChangeSelected(DESTINATION_PREVIOUS)}
      >
        <Icon rotatable size="md">
          &#10094;
        </Icon>
      </ArrowButton>
      <ModalContent
        onMouseEnter={props.onMouseEnter}
        onMouseLeave={props.onMouseLeave}
        color={props.bgColor}
      >
        {props.children}
      </ModalContent>
    </ModalWindow>
  </Backdrop>
);

export default Modal;
