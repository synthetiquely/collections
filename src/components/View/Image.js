import glamorous from 'glamorous';

const Image = glamorous.img(
  {
    opacity: '0',
    width: '100%',
    verticalAlign: 'bottom',
    willChange: 'opacity',
    transition: 'opacity 2s',
  },
  ({ loaded }) => {
    if (loaded) {
      return {
        opacity: '1',
      };
    }
    return {};
  },
);

export default Image;
