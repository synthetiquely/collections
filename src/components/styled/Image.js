import glamorous from 'glamorous';

const Image = glamorous.img(
  {
    opacity: '0',
    width: '100%',
    maxHeight: '90vh',
    verticalAlign: 'bottom',
    willChange: 'opacity',
    transition: 'opacity 1s',
  },
  ({ loaded }) => {
    let styles = {};
    if (loaded) {
      styles = {
        ...styles,
        opacity: '1',
      };
    }

    return styles;
  },
);

export default Image;
