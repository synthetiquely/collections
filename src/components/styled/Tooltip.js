import glamorous from 'glamorous';

const Tooltip = glamorous.div(
  {
    position: 'absolute',
    bottom: '75px',
    left: '-75px',
    width: '300px',
    height: '60px',
    opacity: '0',
    willChange: 'opacity, left',
    backgroundColor: 'rgba(0,0,0,0.7)',
    transition: 'all 0.3s ease-in-out',
  },
  ({ showTooltip }) => {
    let styles = {};
    if (showTooltip) {
      styles = {
        ...styles,
        opacity: '1',
        left: '45px',
      };
    }
    return styles;
  },
);

export default Tooltip;
