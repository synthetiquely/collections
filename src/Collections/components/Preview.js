import React from 'react';
import glamorous from 'glamorous';

import Modal from '../../Common/Modal';

const Image = glamorous.img({
  width: '100%',
  height: '100%',
  objectFit: 'cover',
});

export default ({ src, title, ...rest }) => (
  <Modal {...rest}>
    <Image src={src} alt={title} />
  </Modal>
);
