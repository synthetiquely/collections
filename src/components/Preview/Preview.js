import React, { Component } from 'react';
import { observer } from 'mobx-react';
import Modal from '../Modal/Modal';
import Swipe from '../Swipe/Swipe';
import Image from '../styled/Image';
import Tooltip from '../styled/Tooltip';
import TooltipText from '../styled/TooltipText';

import {
  DESTINATION_PREVIOUS,
  DESTINATION_NEXT,
  KEY_ARROW_LEFT,
  KEY_ARROW_RIGHT,
  KEY_ESC,
} from '../../constants';

@observer
class Preview extends Component {
  constructor(props) {
    super(props);
    this.state = { showTooltip: false, loaded: false };
    this.onLoad = this.onLoad.bind(this);
    this.onSwipe = this.onSwipe.bind(this);
    this.onMouseMove = this.onMouseMove.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);
    this.onClose = this.onClose.bind(this);
    this.onChangeSelected = this.onChangeSelected.bind(this);
  }

  componentDidMount() {
    document.body.addEventListener('keydown', this.onKeyDown);
  }

  componentDidUpdate() {
    if (this.props.collections.selectedPhoto) {
      document.body.classList.add('overlayed');
    } else {
      document.body.classList.remove('overlayed');
    }
  }

  componentWillUnmount() {
    document.body.removeEventListener('keydown', this.onKeyDown);
  }

  onLoad() {
    this.setState({
      loaded: true,
    });
  }

  onChangeSelected(destination) {
    this.props.collections.slideNext(destination);
  }

  onClose() {
    this.props.collections.clearSelection();
    this.setState({
      loaded: false,
      showTooltip: false,
    });
  }

  onKeyDown(e) {
    if (e.keyCode) {
      // eslint-disable-next-line
      switch (e.keyCode) {
        case KEY_ARROW_LEFT:
          this.onChangeSelected(DESTINATION_PREVIOUS);
          break;
        case KEY_ARROW_RIGHT:
          this.onChangeSelected(DESTINATION_NEXT);
          break;
        case KEY_ESC:
          this.onClose();
      }
    }
  }

  onMouseMove() {
    this.setState(prevState => ({
      showTooltip: !prevState.showTooltip,
    }));
  }

  onSwipe(direction) {
    if (direction === 'left' || direction === 'down') {
      this.onChangeSelected(DESTINATION_PREVIOUS);
    } else if (direction === 'right' || direction === 'up') {
      this.onChangeSelected(DESTINATION_NEXT);
    }
  }

  render() {
    const { showTooltip, loaded } = this.state;

    if (!this.props.collections.selectedPhoto) {
      return null;
    }

    return (
      <Modal
        bgColor={!loaded && this.props.collections.selectedPhoto.color}
        {...this.props}
        onClose={this.onClose}
        onChangeSelected={this.onChangeSelected}
      >
        <Swipe onSwipe={this.onSwipe}>
          <a href={this.props.collections.selectedPhoto.url} target="_blank">
            <Image
              style={{ width: 'auto', maxWidth: '100%', objectFit: 'contain' }}
              loaded={loaded}
              src={this.props.collections.selectedPhoto.fullsrc}
              alt={this.props.collections.selectedPhoto.description}
              onLoad={this.onLoad}
              onMouseEnter={this.onMouseMove}
              onMouseLeave={this.onMouseMove}
            />
            {loaded && (
              <Tooltip id="preview-tooltip" showTooltip={showTooltip && loaded}>
                <TooltipText>
                  {this.props.collections.selectedPhoto.user.username}
                </TooltipText>
              </Tooltip>
            )}
          </a>
        </Swipe>
      </Modal>
    );
  }
}

export default Preview;
