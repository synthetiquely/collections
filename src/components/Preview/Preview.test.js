import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Preview from './Preview';
import Modal from '../Modal/Modal';
import Image from '../styled/Image';
import Tooltip from '../styled/Tooltip';

describe('Preview Component', () => {
  const props = {
    collections: {
      selectedPhoto: {
        height: 300,
        width: 250,
        color: 'red',
        fullsrc:
          'https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg',
        description: 'Test Image',
        url: 'url',
        user: {
          username: 'test user',
        },
      },
      slideNext: jest.fn(),
      clearSelection: jest.fn(),
    },
  };
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<Preview {...props} />);
    const tree = renderer.create(<Preview {...props} />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render an image, if there is a seleted photo in the store', () => {
    const wrapper = mount(<Preview {...props} />);

    expect(wrapper.find(Modal)).toHaveLength(1);
    expect(wrapper.find(Image)).toHaveLength(1);
    expect(wrapper.find(Image).props()).toHaveProperty(
      'src',
      props.collections.selectedPhoto.fullsrc,
    );
    expect(wrapper.find(Tooltip)).toHaveLength(1);
  });

  it('should render nothing, if there is no seleted photo', () => {
    const testProps = {
      ...props,
      collections: {
        ...props.collections,
        selectedPhoto: null,
      },
    };
    const wrapper = mount(<Preview {...testProps} />);

    expect(wrapper.find(Modal)).toHaveLength(0);
  });
});
