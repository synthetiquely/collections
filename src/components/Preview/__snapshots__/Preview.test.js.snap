// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Component should render and match its own snapshot 1`] = `
<div
  className="css-3hsllp"
>
  <div
    className="css-13la5a5"
    id="preview-modal"
  >
    <button
      className="css-hcjupp"
      id="preview-close-button"
      onClick={[Function]}
    >
      <span
        className="css-16yzkgv"
      >
        ×
      </span>
    </button>
    <button
      className="css-10e1v19"
      id="preview-right-arrow"
      onClick={[Function]}
    >
      <span
        className="css-937lsz"
      >
        ❯
      </span>
    </button>
    <button
      className="css-1cvb4bz"
      id="preview-left-arrow"
      onClick={[Function]}
    >
      <span
        className="css-937lsz"
      >
        ❮
      </span>
    </button>
    <div
      className="css-186ako1"
      onMouseEnter={undefined}
      onMouseLeave={undefined}
    >
      <div
        className="css-zmgijg"
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <a
          href="url"
          target="_blank"
        >
          <img
            alt="Test Image"
            className="css-eob8a0"
            onLoad={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg"
            style={
              Object {
                "maxWidth": "100%",
                "objectFit": "contain",
                "width": "auto",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
